substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JfrTest.java

```diff
@@ -127,6 +140,44 @@ protected void checkRecording() throws AssertionError {
             Assert.fail("Failed to parse recording: " + e.getMessage());
         }
     }
+
+    private static class ChronologicalComparator implements Comparator<RecordedEvent> {
+        @Override
+        public int compare(RecordedEvent e1, RecordedEvent e2) {
+            return e1.getEndTime().compareTo(e2.getEndTime());
+        }
+    }
+
+    private Path makeCopy(String testName) throws IOException { // from jdk 19
+        Path p = recording.getDestination();
+        if (p == null) {
+            File directory = new File(".");
+            p = new File(directory.getAbsolutePath(), "recording-" + recording.getId() + "-" + testName + ".jfr").toPath();
+            recording.dump(p);
+        }
+        return p;
+    }
+
+    protected List<RecordedEvent> getEvents(String testName) throws IOException {
```

Comment:
I changed that method so that it will filter out all events except the ones that are on the list of tested events. This should allow each test to skip that step.